packages/storage/src/indexed-db/index.ts

```diff
@@ -87,6 +95,18 @@ export class IndexedDb implements IndexedDbInterface {
     return this.u.subscribe(table);
   }
 
+  async *iterateQuery<DBTypes extends PenumbraDb, StoreName extends StoreNames<DBTypes>>(
+    table: StoreName,
+  ) {
+    const tx = this.db.transaction(table);
+    let cursor = await tx.store.openCursor();
+
+    while (cursor) {
+      yield cursor.value;
+      cursor = await cursor.continue();
+    }
+  }
+
```

Comment:
Ah, this kinda relates to https://github.com/penumbra-zone/web/issues/312 right? From this issue, there is a link to: https://github.com/jakearchibald/idb#idbcursor-enhancements. It looks like we can just do `for await` directly on the store?

We may not need this helper then?

```typescript
const tx = db.transaction(storeName);

for await (const cursor of tx.store) {
  // …
}
```

Reply:
Yeah, that turned out to be easier, we can just iterate over stores